### PR TITLE
update hibp to use real range API

### PR DESCRIPTION
**Pull request by groovecoder**
* Adds `HIBP_STAGE` env vars
* Adds handlebars template helpers to format the real HIBP breach metadata
* Updates `hibp` module to call the real stage endpoint for email hash range queries
* Removes `get-hashsets` script as we won't use it anymore

### CODE REVIEW

File: hbs-helpers.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+const HBSHelpers = {
+  init(hbs) {
+
+    hbs.registerHelper("prettyDate", function(date, options) {
```
Comment (pdehaan) [inline hbs-helpers.js]: Me thinks `options` is unused.
This also feels a bit too locale specific. What about something like `toLocaleDateString()`?

```js
return new Date(date).toLocaleDateString('en-US', {year: 'numeric', month: 'long', day: 'numeric'});
```

File: hibp.js [hunk 2]
```diff
@@ -17,25 +17,23 @@ const HIBP = {
 
     const sha1 = getSha1(email);
     const sha1Prefix = sha1.slice(0, 6);
-    const url = `${AppConstants.HIBP_STUB_API_ROOT}/breachedaccount/range/${sha1Prefix}`;
+    const url = `${AppConstants.HIBP_STAGE_API_ROOT}/range/${sha1Prefix}?code=${AppConstants.HIBP_STAGE_API_TOKEN}`;
```
Comment (pdehaan) [inline hibp.js]: I'm guessing the `HIBP_STAGE_API_TOKEN` is just a basic alpha-numeric string and doesn't need to be special URL-encoded?

**Comment by groovecoder** [comment]
Addressed comments, and added 0d99beb7bfaf1943ebe5aad8ed91ecfcdb8694e3 to replace the last use of `popsicle` with `got` so we're using `got` everywhere. (I removed the last use of `request` when I removed the `get-hashsets` script.)

File: package.json [hunk 3]
```diff
@@ -22,8 +21,7 @@
     "nodemailer-express-handlebars": "^3.0.0",
     "objection": "^1.1.9",
     "pg": "^7.4.3",
-    "popsicle": "^9.2.0",
-    "request": "^2.86.0"
+    "popsicle": "^9.2.0"
```
Comment (nhnt11) [inline package.json]: Can we get rid of popsicle from package.json now?

File: views/scan.hbs [hunk 4]
```diff
@@ -63,10 +63,11 @@
...
           <p>
-            <span class="bold">Breach date:</span> {{ meta.date }}<br>
-            <span class="bold">Compromised data:</span> {{ meta.dataClasses }}<br>
-            <span class="bold">Compromised accounts:</span> {{ meta.acCount }}<br>
-            {{ meta.description }}
+            <span class="bold">Breach date:</span> {{prettyDate meta.BreachDate }}<br>
+            <span class="bold">Compromised data:</span> {{breachDataClasses meta.DataClasses }}<br>
+            <span class="bold">Compromised accounts:</span> {{localeString meta.PwnCount }}<br>
+            {{! TODO: make sure meta.Description is a safe string so we can use triple-{{{ }}
+            {{{ meta.Description }}}
```
Comment (pdehaan) [inline views/scan.hbs]: Should we use something like [**dompurify**](https://github.com/cure53/DOMPurify) to sanitize the `meta.Description`?

File: package.json (hunk 3 above)
Comment (groovecoder) [reply package.json]: oh, yeah sorry I meant to do that.

File: views/scan.hbs (hunk 4 above)
Comment (groovecoder) [reply views/scan.hbs]: We should. Should it be part of this PR or a new issue?

Comment (pdehaan) [reply views/scan.hbs]: I'll file a new issue, but we should probably address before release.

**Comment by groovecoder** [comment]
Merging. I'll fix up stage server when it deploys. It will need new required env vars.

**Pull request merged**